test/test_transforms_v2_refactored.py

```diff
@@ -2208,9 +2180,96 @@ def new_resize(dp, *args, **kwargs):
         t(torch.rand(3, 10, 10)).shape == (3, 224, 224)
         t(datapoints.Image(torch.rand(3, 10, 10))).shape == (3, 224, 224)
 
-    def test_bad_disaptcher_name(self):
-        class CustomDatapoint(datapoints.Datapoint):
-            pass
+    def test_errors(self, mocker):
+        mocker.patch.dict(_KERNEL_REGISTRY, clear=True)
 
         with pytest.raises(ValueError, match="Could not find dispatcher with name"):
-            F.register_kernel("bad_name", CustomDatapoint)
+            F.register_kernel("bad_name", datapoints.Image)
+
+        with pytest.raises(ValueError, match="Kernels can only be registered on dispatchers"):
+            F.register_kernel(datapoints.Image, F.resize)
+
+        with pytest.raises(ValueError, match="Kernels can only be registered for subclasses"):
+            F.register_kernel(F.resize, object)
+
+        F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+        with pytest.raises(ValueError, match="already has a kernel registered for type"):
+            F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+
+class TestGetKernel:
+    def make_and_register_kernel(self, dispatcher, input_type):
+        return _register_kernel_internal(dispatcher, input_type, datapoint_wrapper=False)(object())
+
+    @pytest.fixture
+    def dispatcher_and_kernels(self, mocker):
+        mocker.patch.dict(_KERNEL_REGISTRY, clear=True)
+
+        dispatcher = object()
+
+        kernels = {
+            cls: self.make_and_register_kernel(dispatcher, cls)
+            for cls in [
+                torch.Tensor,
+                PIL.Image.Image,
+                datapoints.Image,
+                datapoints.BoundingBoxes,
+                datapoints.Mask,
+                datapoints.Video,
+            ]
+        }
+
+        yield dispatcher, kernels
+
+    def test_unsupported_types(self, dispatcher_and_kernels):
+        dispatcher, _ = dispatcher_and_kernels
+
+        class MyTensor(torch.Tensor):
+            pass
+
+        class MyPILImage(PIL.Image.Image):
+            pass
+
+        for input_type in [str, int, object, MyTensor, MyPILImage]:
+            with pytest.raises(TypeError, match=re.escape(str(input_type))):
```

Comment:
can we catch more precise error message, so we know "why" this is expected to fail when reading the test?